java-checks/src/main/java/org/sonar/java/checks/SwitchLastCaseIsDefaultCheck.java

```diff
@@ -51,12 +48,7 @@ public void visitNode(Tree tree) {
     }
     SwitchStatementTree switchStatementTree = (SwitchStatementTree) tree;
     Optional<CaseLabelTree> defaultLabel = getDefaultLabel(switchStatementTree);
```

Comment:
You should go one step more in simplifying the rule: Change this method to return a boolean: `hasNoDefault(switchStatementTree)`

```
return allLabels(switchStatementTree).noneMatch(SwitchLastCaseIsDefaultCheck::isDefault);
```